src/cryptography/hazmat/backends/openssl/backend.py

```diff
@@ -686,6 +776,68 @@ def cmac_algorithm_supported(self, algorithm):
     def create_cmac_ctx(self, algorithm):
         return _CMACContext(self, algorithm)
 
+    def create_x509_csr(self, builder, private_key, algorithm):
+        # TODO: check type of private key parameter.
```

Comment:
Are you saying this PR should wait until another refactoring occurs or should we add the tests to make the EC and DSA cases work in this PR first?